src/PatternBuilder.ts

```diff
@@ -47,116 +47,126 @@ type TermSequence = { tag: 'termSequence'; terms: ReadonlyArray<Term> }
 
 type Term = Atom | QuantifiedAtom
 
-type QuantifiedAtom = { tag: 'quantifiedAtom'; atom: Atom; greedy: boolean } & (
-  | { kind: 'star' }
-  | { kind: 'plus' }
+type QuantifiedAtom = { tag: 'quantifiedAtom'; atom: Atom } & (
+  | { kind: 'star'; greedy: boolean }
+  | { kind: 'plus'; greedy: boolean }
```

Comment:
While the others can _technically_ be marked as non-greedy, I've never found a reason to do so in practice, which is why I left them off of their corresponding constructor functions. Moving this codifies that into the type to allow 100% test coverage

Reply:
Awesome 👏